src/Engine/UnitMods/UnitModifierWithDefaults.java

```diff
@@ -0,0 +1,53 @@
+package Engine.UnitMods;
+
+import Engine.Combat.CombatContext;
+import Engine.Combat.StrikeParams;
+import Engine.Combat.StrikeParams.BattleParams;
+import Units.UnitContext;
+
+/**
+ * Defaults all the interface items to no-ops for ease of implementation
+ */
+public interface UnitModifierWithDefaults extends UnitModifier
```

Comment:
Why introduce a new class to implement `UnitModifier` instead of just giving `UnitModifier` the default implementations?

Reply:
I'd like to have the option of getting the compiler to tell me whether I've implemented all the methods in this interface or not.
This is relevant for the filter classes, for example.

Reply:
Do you ever (or ever plan to) implement every interface method in the same subclass (excepting `UnitModifierWithDefaults` and `UnitModFilter`, of course)? Those should be stable once we stabilize the set of modifiable things, so I'm not too worried. I guess it doesn't hurt anything either.